ciao-controller/api/api.go

```diff
@@ -95,10 +96,20 @@ func errorResponse(err error) Response {
 // and pass some package level context into the handler.
 type Handler struct {
 	*Context
-	Handler func(*Context, http.ResponseWriter, *http.Request) (Response, error)
+	Handler    func(*Context, http.ResponseWriter, *http.Request) (Response, error)
+	Privileged bool
 }
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// check whether we should send permission denied for this route.
+	if h.Privileged {
+		_, ok := service.GetPrivilege(r.Context())
```

Comment:
Are we checking the correct bool here?  Ok is true if the value associated with PrivKey is of type bool.  We don't actually check to see whether that bool is true or not.

It might be simpler to remove the check in the type assertion in  GetPrividege so there is only one bool

```
func GetPrivilege(ctx context.Context) (bool, bool) {
	return  ctx.Value(PrivKey).(bool)
}
```

Not this will obviously panic if the value is not a boot, but if this panic occurs there's something badly wrong with controller and a panic is probably the right thing to do, like dereferencing null.